src/main/webapp/blobstoreFunctions.js

```diff
@@ -0,0 +1,39 @@
+/** Loads action URL and attaches it to the upload form (the user's browser uploads the file to blobstore with the generated url) */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url').then((response) => response.text()).then((imageUploadUrl) => {
+    document.getElementById('plant-pic-form').action = imageUploadUrl;
+  });
+}
+ 
+ 
+/**
+  * Creates and adds a card element for the photo 
+  * pic is the URL of image to be shown in the post
+*/
+function createListElement(pic) {
+  let card = document.createElement("div");
+  let body = document.createElement("div");
+  let photo = document.createElement("img");
+  photo.style.width = "100px";
+ 
+  card.className = "card m-1";
+  body.className = "card-body";
+  photo.src = pic;
```

Comment:
i would leave this as pic_url instead of just pic, since its more descriptive

Reply:
got it, done!